src/Network.cpp

```diff
@@ -325,25 +342,41 @@ std::pair<int, int> Network::load_network_file(std::string filename) {
         myprintf("Could not open weights file: %s\n", filename.c_str());
         return {0, 0};
     }
-
-    // Read format version
-    auto line = std::string{};
-    if (std::getline(wtfile, line)) {
-        auto iss = std::stringstream{line};
-        // First line is the file format version id
-        iss >> m_format_version;
-        if (iss.fail()
-            || m_format_version > MAX_FORMAT_VERSION
-            || m_format_version < 1) {
-            myprintf("Weights file is the wrong version.\n");
+    // Check for gzip id bytes. 31, 139.
+    if (wtfile.get() != 31) {
+        wtfile.unget();
+    } else if (wtfile.get() != 139) {
+        wtfile.unget();
+        wtfile.unget();
+    } else {
+        // File is gzipped, reopen with zlib.
+        wtfile.close();
+        auto gzhandle = gzopen(filename.c_str(), "rb");
```

Comment:
I think gzopen works for uncompressed files too?  That way we can simplify this codepath and not have to check for gz explicitly.

Reply:
Yes that is apparently true.  I'll make an update later.